### ISSUE TITLE
html_url, empty path not normalized

### ISSUE THREAD

**Issue by markov2** [sysread/URI-Fast#41]
The following two produce unchanged (hence different) results but mean the same.  I think the former is the right choice:
```bash
perl -MURI::Fast=html_url -we 'print html_url("http://example.com/",  "http://a")'
perl -MURI::Fast=html_url -we 'print html_url("http://example.com",  "http://a")'
```
I don't care which one gets preference, as long as they normalize to the same.

**Comment by sysread** [comment]
RFC 3986 5.2.3 says:

```o  If the base URI has a defined authority component and an empty
      path, then return a string consisting of "/" concatenated with the
      reference's path; otherwise,```

That does seem to support a trailing slash.

**Comment by sysread** [comment]
Hmph. But reading the prior section, there is different behavior defined for a rel with a defined path (e.g., a trailing `/`) and one without:
```
               if (R.path starts-with "/") then
                  T.path = remove_dot_segments(R.path);
               else
                  T.path = merge(Base.path, R.path);
                  T.path = remove_dot_segments(T.path);
               endif;
```

**Comment by markov2** [comment]
I think the latter is nothing special: it explains the difference between relative paths which start with a slash (relative to the top of the domain) and without slash (relative to the page base).  It does not say anything about whether you need to write `http://example.com` or `http://example.com/`

**Comment by markov2** [comment]
Yes, AFAIK, there should be a trailing slash in the normalized/canonical version.  `URI` adds the slash.

**Comment by sysread** [comment]
I think you are right. This seems to confirm it: https://datatracker.ietf.org/doc/html/rfc3986#section-6.2.3

**Comment by sysread** [comment]
Fixed in v0.53.
